src/NServiceBus.Core/Routing/MessageEndpointMappingCollection.cs

```diff
@@ -120,7 +120,14 @@ public void Add(MessageEndpointMapping mapping)
         /// </summary>
         protected override void BaseAdd(ConfigurationElement element)
         {
-            BaseAdd(element, true);
+            try
+            {
+                BaseAdd(element, true);
+            }
+            catch (ConfigurationErrorsException e)
+            {
+                throw new Exception($"An ambigious message endpoint mapping has been defined at line: {e.Line}. Check the 'MessageEndpointMappings' section in {e.Filename} for conflicting mappings.", e);
```

Comment:
the property goes over the indexer of ConfigurationElement https://github.com/Particular/NServiceBus/blob/9945aee14e2d530a51d18b3ec9b5f09be59a4d98/src/NServiceBus.Core/Routing/MessageEndpointMapping.cs#L28 which uses the magic of configuration element to convert values. This can fail. Anyway I brought up my points. You both seem to disagree. Let's move on.